docs/tutorial/frontend-tut.md

```diff
@@ -1,94 +1,117 @@
-Since the `reduce` parallelism factor is also 1, the reduction is performed sequentially.
+## Example: sum of squares
+
+Here's a MrXL program that squares and then sums the values of an input array:
+```
+{{#include ../../frontends/mrxl/test/sos.mrxl}}
+```
+
+This short program shows off all of MrXL's features, so let's pick it apart line by line:
+1. We specify an array, `avec`, which will have four integers. The `input` keyword means that an external harness will populate the array.
+2. We specify `sos`, a register. The `output` keyword means that we will populate `sos` in our program.
+3. The `map` operation gets the values of `avec` and raises each to the second power. We stash the result in a new array, `squares`. The number `1` denotes a *parallelism factor* of 1, meaning that the operation is performed sequentially. We will improve this shortly.
+4. The `reduce` operation walks over `squares` and accumulates the result into a register. The parallelism factor is again `1`.
 
 
-## Run a MrXL Program
+## Running our example
 
-Once we have [installed the mrxl command line tool][mrxldocs], we can run MrXL programs using [`fud`][fud].
+Let's run this program.
 
-To provide MrXL program with input values, we use fud's [JSON][json]-based [data format][fud-data].
-Let's try to run this program, which has a parallelism factor of two:
+To begin, [install the MrXL command line tool][mrxldocs-install].
+(TK: I'd kinda like to inline those instructions here, so that there is less jumping around between files.
+I want folks to just install and get back here so we can interpret together, but there's a risk that folks will just begin to follow the other guide and go all the way to Verilog there.
+Also, the MrXL tutorial has a different running example.
+Thoughts on the inline?)
 
+Now change directories to `calyx/frontends/mrxl` and run:
 ```
-{{#include ../../frontends/mrxl/test/add.mrxl}}
+mrxl test/sos.mrxl --data test/sos.mrxl.data --interpret
 ```
-In order to take advantage of the parallelism in the program, the MrXL compiler automatically partitions the input memory `foo` into two different *physical banks*: `foo_b0` and `foo_b1`.
-Therefore, we split up our logical `foo` input of `[1, 2, 3, 4]` into `[1,2]` and `[3,4]`:
+
+Why `42`? Because we populated `avec` with:
 ```json
-{{#include ../../frontends/mrxl/test/add.mrxl.data:2:23}}
+{{#include ../../frontends/mrxl/test/sos.mrxl.data}}
 ```
-Our complete data file similarly splits up the input for `baz`.
+and $0^2 + 1^2 + 4^2 + 5^2 = 42$.
+
 
-Run the program with the complete data by typing:
+## Compiling our example into Calyx
+
+Above, we merely _interpreted_ MrXL code in software using a simple, pre-written interpreter implemented in Python.
+Our goal in this tutorial is to build a compiler from MrXL to hardware by translating it to the Calyx IL.
+The Calyx IL code generated by compiling this program looks more like:
+```
+{{#include ./sos.calyx}}
```

Comment:
Too big a code example–it breaks the flow of the text. Maybe there is a way to shortened it/show just a snippet?

Reply:
Agreed. I was thinking of doing some Markdown hackery to limit the height of the block, which would, in theory, introduce a little scrollable block of text. Do you know anything about this? And, if implementable, would that be okay?

Reply:
Hm, I think I found an answer, but it's hard for me to try since I'm not building the doc. https://bookdown.org/yihui/rmarkdown-cookbook/hook-scroll.html#hook-scroll, or Section 7.4 of the same book. Would you mind giving it a shot?

Reply:
@anshumanmohan, FWIW it's not too hard to build the docs yourself. Just do `cargo install mdbook` and then `mdbook serve --open`.

Reply:
This would be super useful, but sadly I can only get it to render a blank page. Sorry if I'm missing something! I'm running those two lines from the `calyx/` directory...

Reply:
That should do it. Can you include the output from `mdbook` in case there's an error? (Feel free to move to Slack if you prefer.)

Reply:
The book's suggestion did not translate to mdBook. Used a `details/summary` thingy to make this work!